ComfortZoneApp/app/src/main/java/com/example/comfortzone/flight/ui/FlightFragment.java

```diff
@@ -186,12 +176,14 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
 
             @Override
             public void afterTextChanged(Editable s) {
+                FilteringUtils.comfortLevelFilter(rsComfortFilter, cityList, db);
                 if (!s.toString().isEmpty()) {
                     String city = Normalizer.normalize(s, Normalizer.Form.NFD);
                     city = city.replaceAll("[^\\p{ASCII}]", "");
-                    List<WeatherData> searchedCity = FilteringUtils.searchCity(city, cityList);
-                    updateViewsList(searchedCity);
+                    cityList = FilteringUtils.searchCity(city, cityList);
```

Comment:
you probably want to search from the non-shrink list, so we will need a separate list to hold the no-search filtered list